File: codeforces/1800~1899/1862/a.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0, m = 0;
  cin >> n >> m;
  vector<string> carpet(n);
  for (int i = 0; i < n; ++i) cin >> carpet[i];

  string v = "vika";
  for (int j = 0, k = 0; j < m; ++j) {
    for (int i = 0; i < n; ++i) {
      if (carpet[i][j] == v[k]) {
        k++;
        break;
      }
    }
    if (k == (int) v.length()) {
      cout << "YES" << endl;
      return;
    }
  }

  cout << "NO" << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1862/a.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn read_str() -> String {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let inputs = read_vec::<usize>();
    let n = inputs[0];
    let m = inputs[1];
    let mut carpet: Vec<String> = Vec::new();
    for _i in 0..n {
        carpet.push(read_str());
    }
    let v = "vika";
    let mut k: usize = 0;
    for j in 0..m {
        for i in 0..n {
            if carpet[i].as_bytes()[j] == v.as_bytes()[k] {
                k += 1;
                break;
            }
        }

        if k == v.len() {
            writeln!(writer, "YES").ok();
            return;
        }
    }

    writeln!(writer, "NO").ok();
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
